kubernetes/watcher.go

```diff
@@ -217,6 +229,16 @@ func (w *watcher) enqueue(obj interface{}, state string) {
 	w.queue.Add(&item{key, obj, state})
 }
 
+// deltaobjects creates a slice with the old and the new version of cache objects that are ready to chane on update events
+// returns a delta struct to the watcher. w.delta[0] is the old version and w.delta[1] the new updated one
+func (w *watcher) deltaobjects(o interface{}, n interface{}) {
```

Comment:
can we replace the return object here to return a struct instead of an array of objects?

Something like

```golang
type Delta struct {
    old runtime.Object
    new runtime.Obect
}
```

will make the code easier to read and require less comments to figure out what is slice[0] vs slice[1]

Reply:
Done with the struct. Let me know if is ok

Reply:
@gsantoro due to changes https://github.com/elastic/elastic-agent-autodiscover/pull/71#discussion_r1446044940 the struct is not needed anymore. I only return old object before update